include/gsl/util

```diff
@@ -52,40 +52,29 @@ template <class F>
 class final_action
 {
 public:
-    static_assert(!std::is_reference<F>::value && !std::is_const<F>::value &&
-                      !std::is_volatile<F>::value,
-                  "Final_action should store its callable by value");
+    template <class FF>
+    explicit final_action(FF&& ff) noexcept : f{std::forward<FF>(ff)} { }
 
-    explicit final_action(F f) noexcept : f_(std::move(f)) {}
+    ~final_action() noexcept { if (invoke) f(); }
 
     final_action(final_action&& other) noexcept
-        : f_(std::move(other.f_)), invoke_(std::exchange(other.invoke_, false))
-    {}
+        : f(std::move(other.f)), invoke(std::exchange(other.invoke, false))
```

Comment:
`F` might be an lvalue reference to non-`const`, in which case the initialization `f(std::move(other.f))` is ill-formed - a non-`const` lvalue reference can't bind to an rvalue. If you truly intend to support `F` as lvalue reference this should be:
```suggestion
        : f(std::forward<F>(other.f)), invoke(std::exchange(other.invoke, false))
```
If, on the other hand, you don't intend `F` to support references, `finally` should return `final_action<std::remove_reference_t<F>>{std::forward<F>(f)}`.

Reply:
There's nothing in the Guidelines about supporting references, so I'll change `finally`. Thanks!